public/app/plugins/datasource/tempo/types.ts

```diff
@@ -36,6 +36,7 @@ export type TempoQueryType = 'traceql' | 'search' | 'traceId' | 'serviceMap' | '
 
 export interface TempoQuery extends DataQuery {
   query: string;
+  traceQLQuery?: string;
```

Comment:
It's better to require the field to have a value, instead of allowing `undefined` in this case. This decision forced you to have to alter the type of the shared monaco code editor, and if we're not careful it may result in bugs in other parts of the codebase that are using it. 

A better solution in this case is to force `traceQLQuery` to never be undefined and edit the `defaults...()` function to assign it an empty string.